<launch>
<group ns='RArm'>
    <node name="dynamixel_manager" pkg="dynamixel_controllers" type="controller_manager.py" required="true" output="screen">
        <rosparam>
            namespace: dxl_manager
            serial_ports:
                conbe_R_port:
                    port_name: "/dev/ttyUSB1"
                    baud_rate: 1000000
                    min_motor_id: 10
                    max_motor_id: 17
                    update_rate: 200
        </rosparam>
    </node>
	<!--[2]  start joint state publisher-->
    <param name="robot_description"
    	command="$(find xacro)/xacro --inorder '$(find rrbot_description)/urdf/conbeR.xacro'" />

	<!-- [3] state publisher of L-->
	<node name="State_Publisher" pkg="conbe" type="state_publisherR.py" /> 

	<node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher">
				<param name="/use_gui" value="false"/>
				<rosparam param="/source_list">[/conbeR/joint_states]</rosparam>
	</node>

	<!--[4-1] start meta controller       -->
	<rosparam file="$(find conbe)/config/jointsR.yaml" command="load"/>
	<node name="controller_spawner" pkg="dynamixel_controllers" type="controller_spawner.py"
		args="--manager=dxl_manager
				--port=conbe_R_port
				joint0_controller                       
				joint1_controller                                 
				joint2_controller                       
				joint3_controller
				joint4_controller                       
				joint5_controller
				joint6_controller                       
				"
		output="screen"/>
	<!--[4-2] Start joints trajectory controller controller -->
	<rosparam file="$(find conbe)/config/joints_trajectory_controllerR.yaml" command="load"/>
	<node name="controller_spawner_meta" pkg="dynamixel_controllers" type="controller_spawner.py"
		args="--manager=dxl_manager
				--type=meta
				conbeR_controller
				joint0_controller                       
				joint1_controller                                           
				joint2_controller                       
				joint3_controller
				joint4_controller                       
				joint5_controller
			"
		output="screen"/>

</group>
</launch>